pos_margin/models/pos_order_line.py

```diff
@@ -22,11 +22,21 @@ class PosOrderLine(models.Model):
     @api.multi
     @api.depends('product_id', 'qty', 'price_subtotal')
     def _compute_multi_margin(self):
-        for line in self:
-            if not line.product_id:
-                line.purchase_price = 0
-                line.margin = 0
-            else:
-                line.purchase_price = line.product_id.standard_price
-                line.margin = line.price_subtotal - (
-                    line.product_id.standard_price * line.qty)
+        for line in self.filtered(lambda x: x.product_id):
```

Comment:
You can use : self.filtered('product_id') directly

Reply:
thanks, I didn't know. 
Done. mark as resolved.